extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java

```diff
@@ -562,13 +563,30 @@ private QueryDefinition initializeQueryDefAndState(final Closer closer)
                                    .orElse(MSQWarnings.DEFAULT_MAX_PARSE_EXCEPTIONS_ALLOWED);
     }
 
-
+    ImmutableMap.Builder<String, Object> taskContextOverridesBuilder = ImmutableMap.builder();
+    taskContextOverridesBuilder
+        .put(
+            MultiStageQueryContext.CTX_ENABLE_DURABLE_SHUFFLE_STORAGE,
+            isDurableStageStorageEnabled
+        ).put(
+            MultiStageQueryContext.CTX_COMPOSED_INTERMEDIATE_SUPER_SORTER_STORAGE,
+            MultiStageQueryContext.isComposedIntermediateSuperSorterStorageEnabled(
+                task.getQuerySpec().getQuery().context()
+            )
+        ).put(
```

Comment:
If the `CTX_COMPOSED_INTERMEDIATE_SUPER_SORTER_STORAGE` only then `CTX_INTERMEDIATE_SUPER_SORTER_STORAGE_MAX_LOCAL_BYTES` should be set no ?

Reply:
I think they can work independently - the local bytes limit just enforces a resource limit on the local storage and the composition storage enables joining local disk and durable storage together.
Currently, the `CTX_INTERMEDIATE_SUPER_SORTER_STORAGE_MAX_LOCAL_BYTES` is defaulted as `Long.MAX_VALUE` and is used in the normal flow as well to keep consistency. WDYT?

Reply:
I think we should throw an error if one sets  CTX_COMPOSED_INTERMEDIATE_SUPER_SORTER_STORAGE and CTX_INTERMEDIATE_SUPER_SORTER_STORAGE_MAX_LOCAL_BYTES is not set. 

What will happen is the users will set this flag thinking stuff is going to work but they would be proved wrong. 
As @paul-rogers, throwing errors is a way for people to force reading documentation ^-^

Reply:
Yes, I agree with the documentation part but setting `CTX_INTERMEDIATE_SUPER_SORTER_STORAGE_MAX_LOCAL_BYTES` is also not enough for composition. The user would also have to set the durable storage setting and all the configurations along with it. And in future, if any memory storage is allowed in composition, then that is needed too. 
So, I thought the documentation for the composition parameter would clearly state what all is needed to configure it properly.